src/components/Filters/index.js

```diff
@@ -89,6 +89,9 @@ class Filters extends React.Component {
       edit: this.generateFreshEdit(),
       searchString: (existingRule && existingRule.value) || ''
     }
+
+    // Clean exsting rules on load
+    this.filterAndSetRules(rules)
```

Comment:
**This might be an extreme case**, but this would not protect us in case the consumer mutates or assigns a new array with invalid rules after the component initializes. Should we add something similar to `componentWillReceiveProps`?

Alternatively, would it make sense to only have sanitization code in the render function, which the single place responsible to distribute the models in the subcomponents?
```js
const rules = this.filterInvalid(this.props.rules || [])
```
PS: I guess that the second will be costly, so let's forget it.

Reply:
I would argue that at that point we've tried our best to help the consumer, if they have bugged code, they are on their own :)